container/multi-value-slice/multi_value_slice.go

```diff
@@ -0,0 +1,382 @@
+package multi_value_slice
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/prysmaticlabs/prysm/v4/container/multi-value-slice/interfaces"
+)
+
+// MultiValueSlice defines an abstraction over all concrete implementations of the generic Slice.
```

Comment:
It might be worth it to split this up and have a PR about this container we can get a cleaner review. Not a strong opinion thought!